src/main/java/net/neoforged/neoforge/event/BuildCreativeModeTabContentsEvent.java

```diff
@@ -61,12 +65,141 @@ public boolean hasPermissions() {
         return this.parameters.hasPermissions();
     }
 
-    public MutableHashedLinkedMap<ItemStack, CreativeModeTab.TabVisibility> getEntries() {
-        return this.entries;
+    /**
+     * The current immutable ordered set of the parent tab entries in the order to be added to the Creative Menu.
+     * Purely for querying to see what in it. Please use the other event methods for modifications.
+     */
+    public ObjectSortedSet<ItemStack> getParentEntries() {
+        return ObjectSortedSets.unmodifiable(this.parentEntries);
+    }
+
+    /**
+     * The current immutable ordered set of the search tab entries in the order to be added to the Creative Menu.
+     * Purely for querying to see what in it. Please use the other event methods for modifications.
+     */
+    public ObjectSortedSet<ItemStack> getSearchEntries() {
+        return ObjectSortedSets.unmodifiable(this.searchEntries);
     }
 
+    /**
+     * Inserts the new stack at the end of the given tab at this point in time.
+     * 
+     * @exception IllegalArgumentException if the new itemstack's count is not 1 or entry already was added to the tab previously.
+     */
     @Override
-    public void accept(ItemStack stack, CreativeModeTab.TabVisibility visibility) {
-        getEntries().put(stack, visibility);
+    public void accept(ItemStack newEntry, CreativeModeTab.TabVisibility visibility) {
+        assertStackCount(newEntry);
+
+        if (isParentTab(visibility)) {
+            assertNewEntryDoesNotAlreadyExists(parentEntries, newEntry);
+            parentEntries.add(newEntry);
+        }
+
+        if (isSearchTab(visibility)) {
+            assertNewEntryDoesNotAlreadyExists(searchEntries, newEntry);
+            searchEntries.add(newEntry);
+        }
+    }
+
+    /**
+     * Inserts the new entry after the specified existing entry.
+     * 
+     * @exception UnsupportedOperationException if the existing entry is not found in the tab's lists.
+     * @exception IllegalArgumentException      if the new itemstack's count is not 1 or target does not exist in set.
```

Comment:
First, use `@throws` instead of the old `@exception` tag. 

Second, only `IllegalArgumentException` is thrown by the two assertion methods.

(Same comments for `putBefore`.)